src/parser.ts

```diff
@@ -3574,6 +3581,8 @@ export function parseAsyncExpression(
     if (inNew) report(parser, Errors.InvalidAsyncArrow);
     return parseArrowFromIdentifier(parser, context, parser.tokenValue, expr, inNew, canAssign, 0, start, line, column);
   }
+
+  parser.assignable = AssignmentKind.Assignable;
```

Comment:
Given your change on line 1114, should this line be added before line 2694?

Reply:
It seems not needed here, added a couple more test-cases to cover it